src/helm/benchmark/annotation/model_as_judge.py

```diff
@@ -23,23 +23,14 @@ def score_with_reasoning(
         raise Exception(f"Annotation request failed: {annotator_response.error}")
     assert len(annotator_response.completions) == 1
     annotator_response_text = annotator_response.completions[0].text
-    json_start_index = annotator_response_text.find("{")
-    json_end_index = annotator_response_text.rfind("}")
-    if json_start_index < 0 or json_end_index < 0:
-        raise Exception(f"Malformed annotator response: {annotator_response_text}")
-    annotator_response_json = annotator_response_text[json_start_index : json_end_index + 1]
-    try:
-        parsed_response = json.loads(annotator_response_json)
-    except Exception as e:
-        raise Exception(f"Malformed annotator response: {annotator_response_text}") from e
 
-    if not parsed_response:
-        raise Exception(f"Malformed annotator response: {annotator_response_text}")
+    # fuzzy match regex check, allows for different casing, or forgetting / in end tag
+    reasoning_match = re.search(r"<reasoning>(.*?)</?\s*reasoning>", annotator_response_text, re.DOTALL | re.IGNORECASE)
+    score_match = re.search(r"<score>(.*?)</?\s*score>", annotator_response_text, re.DOTALL | re.IGNORECASE)
+    reasoning = reasoning_match.group(1).strip() if reasoning_match else "Reasoning not found"
+    score = score_match.group(1).strip() if score_match else None
 
-    try:
-        score = float(parsed_response.get("score"))
-        reasoning = parsed_response.get("reasoning").strip()
-    except ValueError as e:
-        raise Exception(f"Malformed annotator response: {annotator_response_text}") from e
+    if score == None:
+        raise Exception(f"Score not found in: {annotator_response_text}")
 
-    return {"reasoning": reasoning, "score": score}
+    return {"reasoning": reasoning, "score": float(score)}
```

Comment:
Type error:
```
src/helm/benchmark/annotation/model_as_judge.py:36: error: Argument 1 to "float" has incompatible type "Union[str, Any, None]"; expected "Union[SupportsFloat, SupportsIndex, str, Buffer]"  [arg-type]
```

Not sure why this happens, but you could do `assert isinstance(score, str)` or `cast(str, score)` to get around it.

Reply:
Resolved another way